### PR TITLE
Minor optimization to addon-resizer

### PR DESCRIPTION
**Pull request by wojtek-t**
/assign @mm4tt @bskiba

### CODE REVIEW

File: addon-resizer/nanny/kubernetes_client.go [hunk 1]
```diff
@@ -85,28 +89,35 @@ func (k *kubernetesClient) countNodesThroughMetrics() (uint64, error) {
 		return 0, err
 	}
 
-	decoder := expfmt.SampleDecoder{
-		Dec:  expfmt.NewDecoder(strings.NewReader(string(rawMetrics)), expfmt.FmtText),
-		Opts: &expfmt.DecodeOptions{},
-	}
-	var v model.Vector
+	decoder := expfmt.NewDecoder(strings.NewReader(string(rawMetrics)), expfmt.FmtText)
```
Comment (mm4tt) [inline addon-resizer/nanny/kubernetes_client.go:92]: It should be 
```
decoder := expfmt.NewDecoder(bytes.NewReader(rawMetrics), expfmt.FmtText)
```

Calling string(byteBuffer) allocates memory, I wrote this simple benchmark to confirm that - https://github.com/mm4tt/sandbox/blob/master/gobench/bench_test.go#L40 
So calling string here doubles the memory we need to store rawMetrics.

Comment (wojtek-t) [reply addon-resizer/nanny/kubernetes_client.go]: Thanks - though no longer valid after my next change. PTAL

**Comment by mm4tt** [comment]
Nice

/lgtm

**Comment by wojtek-t** [comment]
Tested more - everything seem to work fine

/hold cancel

**Comment by wojtek-t** [comment]
@bskiba - can you please take a look?

**Comment by wojtek-t** [comment]
Renamed - it's not minor any more ;)

**Comment by wojtek-t** [comment]
Thanks for quick review @bskiba !

**Comment by mm4tt** [comment]
FTR, a memory usage graph showing the impact
PdqBnpANbfm

**Pull request merged**